c7n/resources/ebs.py

```diff
@@ -1085,7 +1085,15 @@ def process(self, volumes):
         retry = get_retry(['Throttled'], max_attempts=5)
         for vol in volumes:
             vol_id = vol['VolumeId']
-            retry(client.create_snapshot, VolumeId=vol_id)
+            retry(self.process_volume, client=client, volume=vol_id)
+
+    def process_volume(self, client, volume):
+        try:
+            client.create_snapshot(VolumeId=volume)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'InvalidVolume.NotFound':
+                return
+            raise e
```

Comment:
just do `raise`, which will preserve original stack trace, `raise e` will create a new stack trace from this point.

Reply:
sounds good